src/composables/useClient.ts

```diff
@@ -10,6 +10,7 @@ export function useClient() {
   const { web3 } = useWeb3();
```

Comment:
This component has become a little hard to read. Can we try something like this?

```js
const ACTIONS = {
  proposal: 'proposal',
  vote: 'vote',
  deleteProposal: 'delete-proposal',
  settings: 'settings',
  deleteSpace: 'delete-space'
};

export function useClient() {

  async function executeAndHandleError(func) {
    isSending.value = true;
    try {
      return await func();
    } catch (e: any) {
      errorNotification(e?.error_description || '');
      return e;
    } finally {
      isSending.value = false;
    }
  }

  function bindWithAlias(actionFn) {
    return (client, payload) => {
      return actionWithAlias(aliasWallet =>
        actionFn(client, payload, {
          provider: aliasWallet,
          address: aliasWallet.address
        })
      );
    };
  }

  async function sendEIP712(space, type, payload) {
    const client = isGnosisSafe.value ? clientGnosisSafe : clientEIP712;
    const shouldUseAlias = space.voting?.aliased && !isGnosisSafe.value;

    switch (type) {
      case ACTIONS.proposal:
        return executeAndHandleError(() => sendProposal(space, payload, client, shouldUseAlias));
      case ACTIONS.vote:
        return executeAndHandleError(() => sendVote(space, payload, client, shouldUseAlias));
      case ACTIONS.deleteProposal:
        return executeAndHandleError(() => sendDeleteProposal(space, payload, client, shouldUseAlias));
      case ACTIONS.settings:
        return executeAndHandleError(() => sendSettings(space, payload, client));
      case ACTIONS.deleteSpace:
        return executeAndHandleError(() => sendDeleteSpace(space, client));
      default:
        throw new Error(`Unknown action type: ${type}`);
    }
  }

  function sendProposal(space, payload, client, shouldUseAlias) {
    const actionFn = shouldUseAlias ? bindWithAlias(proposal) : proposal;

    let plugins = {};
    if (Object.keys(payload.metadata?.plugins).length !== 0)
      plugins = payload.metadata.plugins;

    const proposalPayload = {
      space: space.id,
      type: payload.type,
      title: payload.name,
      body: payload.body,
      discussion: payload.discussion,
      choices: payload.choices,
      start: payload.start,
      end: payload.end,
      snapshot: payload.snapshot,
      plugins: JSON.stringify(plugins),
      app: definedApp,
      from: web3.value.account
    };

    return actionFn(client, proposalPayload);
  }

  // Similarly implement sendVote, sendDeleteProposal, sendSettings, and sendDeleteSpace


}
```